client/src/components/ResultPanel/ResultPanel.ts

```diff
@@ -4,63 +4,116 @@ import { Uri, ViewColumn, WebviewPanel, l10n, window } from "vscode";
       title, // Title of the panel displayed to the user
       {
         preserveFocus: true,
         viewColumn: sideResult ? ViewColumn.Beside : ViewColumn.Active,
       }, // Editor column to show the new webview panel in.
-      {}, // Webview options. More on these later.
-    );
-    resultPanel.onDidDispose(
-      ((id) => () => {
-        delete resultPanels[id];
-        resultPanel = undefined;
-      })(resultPanelId),
+      { enableScripts: true }, // Webview options.
     );
-    resultPanels[resultPanelId] = resultPanel;
+    webviewPanel.onDidDispose(() => disposePanel(panelId));
+    resultPanel = { webviewPanel, panelId };
   } else {
     const editor = uri
       ? window.visibleTextEditors.find(
           (editor) => editor.document.uri.toString() === uri.toString(),
         )
       : window.activeTextEditor;
-    if (resultPanel.title !== title) {
-      resultPanel.title = title;
+    if (resultPanel.webviewPanel.title !== title) {
+      resultPanel.webviewPanel.title = title;
     }
-    resultPanel.reveal(
+    panelId = resultPanel.panelId;
+    resultPanel.webviewPanel.reveal(
       sideResult ? ViewColumn.Beside : editor?.viewColumn,
       true,
     );
   }
-  resultPanel.webview.html = html;
+
+  const panelHtml = wrapPanelHtml(html, panelId);
+  resultPanel.webviewPanel.webview.html = panelHtml;
+  setContextValue(resultPanel.panelId, panelHtml);
+};
+
+/**
+ *
+ * */
+const wrapPanelHtml = (html: string, panelId: string): string => {
+  return (
+    html
+      // Inject vscode context into our results html body
+      .replace(
+        "<body ",
+        `<body data-vscode-context='${JSON.stringify({
+          preventDefaultContextMenuItems: true,
+          panelId,
+        })}' `,
+      )
+      // Make sure the html and body take up the full height of the parent
+      // iframe so that the context menu is clickable anywhere on the page
+      .replace(
+        "</head>",
+        `<script language="javascript">
+          if(acquireVsCodeApi){
+            const vscode = acquireVsCodeApi();
+            const panelId = '${panelId}'
+            vscode.setState({panelId});
+          }
+         </script>
+         <style>html,body { height: 100% !important; }</style></head>`,
+      )
+  );
+};
+
+export const deserializeWebviewPanel = async (
+  webviewPanel: WebviewPanel,
+  state: ResultPanelState,
+): Promise<void> => {
+  const panelHtml: string = await getContextValue(state.panelId);
+  resultPanel = { panelId: state.panelId, webviewPanel: webviewPanel };
+  webviewPanel.webview.html = panelHtml;
+  webviewPanel.onDidDispose(() => disposePanel(state.panelId));
+};
+
+export const fetchHtmlFor = async (panelId: string) => {
+  let panelHtml: string = "";
+  panelHtml = await getContextValue(panelId);
+  panelHtml = panelHtml.replace(SCRIPT_REGEX, "");
+
+  return panelHtml;
+};
+
+const disposePanel = (id: string) => {
+  console.log(`disposing panel ${id}`);
```

Comment:
nit: is it intended to go in production?

Reply:
Fixed in https://github.com/sassoftware/vscode-sas-extension/pull/935/commits/1708088bbdaee17dfac035c7896e2b7f1fdaf35a